src/ASM/ASMs2D.h

```diff
@@ -84,23 +85,24 @@ class ASMs2D : public ASMstruct, public ASM2D
     : curve(sc), dof(d), code(c) {}
   };
 
-protected:
+public:
   //! \brief Base class that checks if an element has interface contributions.
-  class InterfaceChecker
+  class InterfaceChecker : public ASM::InterfaceChecker
   {
+  protected:
     const ASMs2D& myPatch; //!< Reference to the patch being integrated
   public:
     //! \brief The constructor initialises the reference to current patch.
     explicit InterfaceChecker(const ASMs2D& pch) : myPatch(pch) {}
     //! \brief Empty destructor.
     virtual ~InterfaceChecker() {}
     //! \brief Returns non-zero if the specified element have contributions.
+    //! \param[in] iel Element number
     //! \param[in] I Index in first parameter direction of the element
     //! \param[in] J Index in second parameter direction of the element
-    virtual short int hasContribution(int I, int J) const;
+    virtual short int hasContribution(int iel, int I, int J, int = -1) const;
```

Comment:
Could we specify in the doc that this is index to the knot vector since `ASMs3D::constrainCorner(int I, ...)` which has a completely different interpretation of "\param[in] I Parameter index in u-direction"